lib/open_api_spex/cast.ex

```diff
@@ -127,9 +127,17 @@ defmodule OpenApiSpex.Cast do
     {:ok, nil}
   end
 
-  # nullable: false
-  def cast(%__MODULE__{value: nil} = ctx) do
-    error(ctx, {:null_value})
+  # nullable not present in root schema or equal to false
+  # dispatch to xxxOf modules if corresponding key is present
+  # or return error
+  def cast(%__MODULE__{value: nil, schema: schema} = ctx) do
+    cond do
+      match?(%{nullable: false}, schema) -> error(ctx, {:null_value})
+      match?(%{oneOf: list} when is_list(list), schema) -> OneOf.cast(ctx)
+      match?(%{anyOf: list} when is_list(list), schema) -> AnyOf.cast(ctx)
+      match?(%{allOf: list} when is_list(list), schema) -> AllOf.cast(ctx)
+      true -> error(ctx, {:null_value})
+    end
```

Comment:
[nit] This imho will read better with the pattern matching happening in the function head.

```elixir
def cast(%__MODULE__{value: nil, schema: %{nullable: false}} = ctx), do: error(ctx, {:null_value})
def cast(%__MODULE__{value: nil, schema: %{oneOf: _}} = ctx), do: OneOf.cast(ctx)
def cast(%__MODULE__{value: nil, schema: %{anyOf: _}} = ctx), do: AnyOf.cast(ctx)
def cast(%__MODULE__{value: nil, schema: %{allOf: _}} = ctx), do: AllOf.cast(ctx)
def cast(%__MODULE__{value: nil} = ctx), do: error(ctx, {:null_value})
```

Reply:
Addressed in latest commit. Slightly changed as it was not functionally equivalent.

Reply:
@lucacorti Do we need the guard `when is_list(list)` here? Shouldn't this be a responsibility of the `XxxOf.cast/1`?